digest.go

```diff
@@ -275,6 +278,14 @@ func (t *Transport) RoundTrip(req *http.Request) (*http.Response, error) {
 		return nil, err
 	}
 
+	// Obtain a fresh body.
```

Comment:
can you refactor this to up above and check first if `req.Body != nil` and then have an if/else for when GetBody is nil or not?

Reply:
The reason I put the GetBody call way down here was to avoid req2.Body.Close calls before each error return (lines 260, 263, 271, and 278). Are you sure about paying that price to be able to keep all body-related code in one place?

Reply:
that's fair, carry on :)